### PR TITLE
siac: add recursive directory upload

### PR DESCRIPTION
**Pull request by lukechampine**
This is untested. I recommend testing on both Windows and Linux to confirm that the path separator doesn't cause problems.

### CODE REVIEW

**Comment by DavidVorick** [comment]
no appveyor?

**Comment by DavidVorick** [comment]
build failed btw

**Comment by lukechampine** [comment]
I abused appveyor a little testing this. Caught at least one bug. Only thing I'm unsure about is converting Windows paths to siapaths. For example, if you upload directory `foo` on Linux, you get:
```
POST /renter/upload/foo/bar
POST /renter/upload/foo/baz
POST /renter/upload/foo/quux
```
but on Windows you get:
```
POST /renter/upload/foo\bar
POST /renter/upload/foo\baz
POST /renter/upload/foo\quux
```
These don't result in an error, but shouldn't they? The existing `upload` command doesn't seem to convert the path at all, so this behavior has existed for a while on Windows with apparently no complaints. It's easy enough to do the conversion, I'm just curious why it hasn't been an issue. @johnathanhowell, can you give any input on how Sia-UI handles this? From a cursory look, I can see you use `posix` filepaths in a couple places.

**Comment by johnathanhowell** [comment]
Sia-UI uses only forward slashes in the calls to `/renter/upload`, parsing the required path by taking the basename of the file to upload and joining it with the current working siapath in the file browser.

**Comment by DavidVorick** [comment]
I think we wanted the api to only accept normal/standard slashes. I might remember that we have safeties on the Sia backend as well, but I think general/best/suggested practice is for the frontend to convert to all forward slashes for siapaths.

But for the sourcefile I think it makes sense to use the native slashes.

File: siac/rentercmd.go [hunk 1]
```diff
@@ -491,6 +491,7 @@ func renterfilesuploadcmd(source, path string) {
 		for _, file := range files {
 			fpath, _ := filepath.Rel(source, file)
 			fpath = filepath.Join(path, fpath)
+			fpath = filepath.ToSlash(fpath)
```
Comment (lukechampine) [inline siac/rentercmd.go]: looking at the source, this seems to do a naive `string.Replace`. I feel like that wouldn't work on escaped path separators, but apparently it's good enough for the stdlib.

**Pull request merged**